diff --git a/chrome/browser/download/download_ui_model.cc b/chrome/browser/download/download_ui_model.cc
index c90c8536e34c14bad1ebc200f442c3b07e6b9fb9..3f2acb0f8fe9636ac4b372979f846793a911f8cd 100644
--- a/chrome/browser/download/download_ui_model.cc
+++ b/chrome/browser/download/download_ui_model.cc
@@ -28,6 +28,7 @@
 #include "chrome/common/url_constants.h"
 #include "chrome/grit/branded_strings.h"
 #include "chrome/grit/generated_resources.h"
+#include "brave/grit/brave_generated_resources.h"
 #include "components/download/public/common/download_danger_type.h"
 #include "components/enterprise/common/proto/connectors.pb.h"
 #include "components/google/core/common/google_util.h"
@@ -615,6 +616,9 @@ bool DownloadUIModel::IsCommandEnabled(
     case DownloadCommands::ALWAYS_OPEN_TYPE:
     case DownloadCommands::OPEN_WITH_MEDIA_APP:
     case DownloadCommands::EDIT_WITH_MEDIA_APP:
+    case DownloadCommands::REMOVE_FROM_LIST:
+    case DownloadCommands::DELETE_LOCAL_FILE:
+    case DownloadCommands::COPY_DOWNLOAD_LINK:
       NOTREACHED();
     case DownloadCommands::CANCEL:
       return !IsDone();
@@ -679,6 +683,9 @@ bool DownloadUIModel::IsCommandChecked(
     case DownloadCommands::CANCEL_DEEP_SCAN:
     case DownloadCommands::OPEN_WITH_MEDIA_APP:
     case DownloadCommands::EDIT_WITH_MEDIA_APP:
+    case DownloadCommands::REMOVE_FROM_LIST:
+    case DownloadCommands::DELETE_LOCAL_FILE:
+    case DownloadCommands::COPY_DOWNLOAD_LINK:
       return false;
   }
   return false;
@@ -754,6 +761,9 @@ void DownloadUIModel::ExecuteCommand(DownloadCommands* download_commands,
       break;
     case DownloadCommands::OPEN_WITH_MEDIA_APP:
     case DownloadCommands::EDIT_WITH_MEDIA_APP:
+    case DownloadCommands::REMOVE_FROM_LIST:
+    case DownloadCommands::DELETE_LOCAL_FILE:
+    case DownloadCommands::COPY_DOWNLOAD_LINK:
 #if BUILDFLAG(IS_CHROMEOS)
       OpenUsingMediaApp();
       break;
@@ -1090,8 +1100,8 @@ DownloadUIModel::BubbleStatusTextBuilder::GetInProgressStatusText() const {
           l10n_util::GetStringUTF16(
               IDS_DOWNLOAD_BUBBLE_STATUS_ASYNC_CHECKING_SAFETY));
     }
-    // "120 MB • Done"
-    return StatusTextBuilderUtils::GetCompletedTotalSizeString(total_bytes);
+    // "120 MB • Finishing…"
+    return get_total_string(l10n_util::GetStringUTF16(IDS_BRAVE_DOWNLOAD_BUBBLE_STATUS_FINISHING));
   }
 }
 
@@ -1099,7 +1109,7 @@ std::u16string
 DownloadUIModel::StatusTextBuilderBase::GetCompletedRemovedOrSavedStatusText()
     const {
   if (model_->GetFileExternallyRemoved()) {
-    return l10n_util::GetStringUTF16(IDS_DOWNLOAD_STATUS_REMOVED);
+    return l10n_util::GetStringUTF16(IDS_DOWNLOAD_FILE_REMOVED);
   }
   return std::u16string();
 }
