diff --git a/chrome/browser/ui/startup/startup_browser_creator_impl.cc b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
index 7eab8f7d297ca7b7b4a7d2bd4a5d988613064aba..52fb14286e631423e677fc0b88e828636952379c 100644
--- a/chrome/browser/ui/startup/startup_browser_creator_impl.cc
+++ b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
@@ -426,6 +426,7 @@ BrowserWindowInterface* StartupBrowserCreatorImpl::OpenTabsInBrowser(
       params.disposition = WindowOpenDisposition::CURRENT_TAB;
       params.tabstrip_add_types = ADD_NONE;
       first_tab = false;
+      BraveModifyStartupTabNavigationParams(tab, browser, params);
       Navigate(&params);
       continue;
     }
@@ -456,6 +457,7 @@ BrowserWindowInterface* StartupBrowserCreatorImpl::OpenTabsInBrowser(
     }
 #endif  // BUILDFLAG(ENABLE_RLZ)
 
+    BraveModifyStartupTabNavigationParams(tab, browser, params);
     Navigate(&params);
     first_tab = false;
   }
@@ -474,7 +476,7 @@ BrowserWindowInterface* StartupBrowserCreatorImpl::OpenTabsInBrowser(
     base::nix::SetActivationToken(startup_id);
   }
 #endif
-  browser->GetWindow()->Show();
+  BraveShowStartupBrowser(browser, tabs);
 
   return browser;
 }
