diff --git a/chrome/browser/download/download_ui_model.cc b/chrome/browser/download/download_ui_model.cc
index 5fd2326b92df7416267e29c23dcf777dac230cac..b86d2e0d3c38457d256ab7cd22351cbe83bef1bf 100644
--- a/chrome/browser/download/download_ui_model.cc
+++ b/chrome/browser/download/download_ui_model.cc
@@ -28,6 +28,7 @@
 #include "chrome/common/url_constants.h"
 #include "chrome/grit/branded_strings.h"
 #include "chrome/grit/generated_resources.h"
+#include "brave/grit/brave_generated_resources.h"
 #include "components/download/public/common/download_danger_type.h"
 #include "components/enterprise/common/proto/connectors.pb.h"
 #include "components/google/core/common/google_util.h"
@@ -617,6 +618,9 @@ bool DownloadUIModel::IsCommandEnabled(
     case DownloadCommands::ALWAYS_OPEN_TYPE:
     case DownloadCommands::OPEN_WITH_MEDIA_APP:
     case DownloadCommands::EDIT_WITH_MEDIA_APP:
+    case DownloadCommands::REMOVE_FROM_LIST:
+    case DownloadCommands::DELETE_LOCAL_FILE:
+    case DownloadCommands::COPY_DOWNLOAD_LINK:
       NOTREACHED();
     case DownloadCommands::CANCEL:
       return !IsDone();
@@ -681,6 +685,9 @@ bool DownloadUIModel::IsCommandChecked(
     case DownloadCommands::CANCEL_DEEP_SCAN:
     case DownloadCommands::OPEN_WITH_MEDIA_APP:
     case DownloadCommands::EDIT_WITH_MEDIA_APP:
+    case DownloadCommands::REMOVE_FROM_LIST:
+    case DownloadCommands::DELETE_LOCAL_FILE:
+    case DownloadCommands::COPY_DOWNLOAD_LINK:
       return false;
   }
   return false;
@@ -756,6 +763,9 @@ void DownloadUIModel::ExecuteCommand(DownloadCommands* download_commands,
       break;
     case DownloadCommands::OPEN_WITH_MEDIA_APP:
     case DownloadCommands::EDIT_WITH_MEDIA_APP:
+    case DownloadCommands::REMOVE_FROM_LIST:
+    case DownloadCommands::DELETE_LOCAL_FILE:
+    case DownloadCommands::COPY_DOWNLOAD_LINK:
 #if BUILDFLAG(IS_CHROMEOS)
       OpenUsingMediaApp();
       break;
@@ -1092,8 +1102,8 @@ DownloadUIModel::BubbleStatusTextBuilder::GetInProgressStatusText() const {
           l10n_util::GetStringUTF16(
               IDS_DOWNLOAD_BUBBLE_STATUS_ASYNC_CHECKING_SAFETY));
     }
-    // "120 MB • Done"
-    return StatusTextBuilderUtils::GetCompletedTotalSizeString(total_bytes);
+    // "120 MB • Finishing…"
+    return get_total_string(l10n_util::GetStringUTF16(IDS_BRAVE_DOWNLOAD_BUBBLE_STATUS_FINISHING));
   }
 }
 
@@ -1101,7 +1111,7 @@ std::u16string
 DownloadUIModel::StatusTextBuilderBase::GetCompletedRemovedOrSavedStatusText()
     const {
   if (model_->GetFileExternallyRemoved()) {
-    return l10n_util::GetStringUTF16(IDS_DOWNLOAD_STATUS_REMOVED);
+    return l10n_util::GetStringUTF16(IDS_DOWNLOAD_FILE_REMOVED);
   }
   return std::u16string();
 }
