diff --git a/chrome/browser/download/download_ui_model.cc b/chrome/browser/download/download_ui_model.cc
index fe9306f5618abf7508cc71dc8e072e582fef6a83..96ae4392665cc8c18a2653edaf8b13c385a4cdcc 100644
--- a/chrome/browser/download/download_ui_model.cc
+++ b/chrome/browser/download/download_ui_model.cc
@@ -28,6 +28,7 @@
 #include "chrome/common/url_constants.h"
 #include "chrome/grit/branded_strings.h"
 #include "chrome/grit/generated_resources.h"
+#include "brave/grit/brave_generated_resources.h"
 #include "components/download/public/common/download_danger_type.h"
 #include "components/enterprise/common/proto/connectors.pb.h"
 #include "components/google/core/common/google_util.h"
@@ -648,6 +649,9 @@ bool DownloadUIModel::IsCommandEnabled(
     case DownloadCommands::ALWAYS_OPEN_TYPE:
     case DownloadCommands::OPEN_WITH_MEDIA_APP:
     case DownloadCommands::EDIT_WITH_MEDIA_APP:
+    case DownloadCommands::REMOVE_FROM_LIST:
+    case DownloadCommands::DELETE_LOCAL_FILE:
+    case DownloadCommands::COPY_DOWNLOAD_LINK:
       NOTREACHED();
     case DownloadCommands::CANCEL:
       return !IsDone();
@@ -712,6 +716,9 @@ bool DownloadUIModel::IsCommandChecked(
     case DownloadCommands::CANCEL_DEEP_SCAN:
     case DownloadCommands::OPEN_WITH_MEDIA_APP:
     case DownloadCommands::EDIT_WITH_MEDIA_APP:
+    case DownloadCommands::REMOVE_FROM_LIST:
+    case DownloadCommands::DELETE_LOCAL_FILE:
+    case DownloadCommands::COPY_DOWNLOAD_LINK:
       return false;
   }
   return false;
@@ -787,6 +794,9 @@ void DownloadUIModel::ExecuteCommand(DownloadCommands* download_commands,
       break;
     case DownloadCommands::OPEN_WITH_MEDIA_APP:
     case DownloadCommands::EDIT_WITH_MEDIA_APP:
+    case DownloadCommands::REMOVE_FROM_LIST:
+    case DownloadCommands::DELETE_LOCAL_FILE:
+    case DownloadCommands::COPY_DOWNLOAD_LINK:
 #if BUILDFLAG(IS_CHROMEOS)
       OpenUsingMediaApp();
       break;
@@ -1123,8 +1133,8 @@ DownloadUIModel::BubbleStatusTextBuilder::GetInProgressStatusText() const {
           l10n_util::GetStringUTF16(
               IDS_DOWNLOAD_BUBBLE_STATUS_ASYNC_CHECKING_SAFETY));
     }
-    // "120 MB • Done"
-    return StatusTextBuilderUtils::GetCompletedTotalSizeString(total_bytes);
+    // "120 MB • Finishing…"
+    return get_total_string(l10n_util::GetStringUTF16(IDS_BRAVE_DOWNLOAD_BUBBLE_STATUS_FINISHING));
   }
 }
 
@@ -1132,7 +1142,7 @@ std::u16string
 DownloadUIModel::StatusTextBuilderBase::GetCompletedRemovedOrSavedStatusText()
     const {
   if (model_->GetFileExternallyRemoved()) {
-    return l10n_util::GetStringUTF16(IDS_DOWNLOAD_STATUS_REMOVED);
+    return l10n_util::GetStringUTF16(IDS_DOWNLOAD_FILE_REMOVED);
   }
   return std::u16string();
 }
