diff --git a/chrome/browser/sessions/session_restore.cc b/chrome/browser/sessions/session_restore.cc
index 684340e2ae22a97ffcbb010bc31c9f7c9433492e..b68ff1a7c995d5a465bbb78c278594515c600f34 100644
--- a/chrome/browser/sessions/session_restore.cc
+++ b/chrome/browser/sessions/session_restore.cc
@@ -865,6 +865,8 @@ class SessionRestoreImpl : public BrowserCollectionObserver {
 #endif
       }
 
+      BraveBeginOriginSpaceWindowRestore(browser, *window);
+
       // 2. Track TYPE_NORMAL browsers.
       if (is_normal_window) {
         has_normal_browser = true;
@@ -946,6 +948,8 @@ class SessionRestoreImpl : public BrowserCollectionObserver {
         chrome::CloseWebContents(browser, active_tab, true);
       }
 
+      BraveFinishOriginSpaceWindowRestore(browser);
+
       // Sanity check: A restored browser should have an active tab.
       // TODO(crbug.com/40662817): Change to DCHECK once we understand
       // why some browsers don't have an active tab on startup.
@@ -1386,6 +1390,7 @@ class SessionRestoreImpl : public BrowserCollectionObserver {
                                : WindowOpenDisposition::NEW_BACKGROUND_TAB;
       params.tabstrip_add_types = add_types;
       is_first_tab = false;
+      BraveModifyStartupTabNavigationParams(startup_tab, browser, params);
       Navigate(&params);
     }
   }
